dynamixel_controllers/src/dynamixel_controllers/joint_trajectory_action_controller.py

```diff
@@ -284,23 +293,24 @@ def process_trajectory(self, traj):
                 # by setting the goal to the current position
                 if self.action_server.is_preempt_requested():
                     msg = 'New trajectory received. Aborting old trajectory.'
-                    multi_packet = {}
-                    
-                    for port, joints in self.port_to_joints.items():
-                        vals = []
+                    if self.cancel_prev_goals_on_preemption:
```

Comment:
In position control mode in Gazebo, there's no clear way to cancel a goal once the command is given.